src/main/java/xyz/thepathfinder/android/Cluster.java

```diff
@@ -98,29 +113,32 @@ protected static Cluster getInstance(String path, PathfinderServices services) {
      * cluster object and update the fields. If the path is associated with a different model type or
      * the json will not parse to a cluster object's required fields it will throw a
      * <tt>ClassCastException</tt>.
+     *
      * @param clusterJson A json object that parses to a cluster.
-     * @param services The pathfinder services object.
+     * @param services    The pathfinder services object.
      * @return A cluster with the specified path.
      * @throws ClassCastException if the requested path is already associated with a
-     *         different {@link Model} type. Also, if the json object doesn't parse to
-     *         a cluster object.
+     *                            different {@link Model} type. Also, if the json object doesn't parse to
+     *                            a cluster object.
      */
     protected static Cluster getInstance(JsonObject clusterJson, PathfinderServices services) {
         boolean canParseToCluster = Cluster.checkClusterFields(clusterJson);
 
-        if(!canParseToCluster) {
+        if (!canParseToCluster) {
             throw new ClassCastException("JSON could not be parsed to a cluster");
```

Comment:
I don't think you're ever supposed to throw `ClassCastException`, at least not in parsing methods. From the Java API: "Thrown to indicate that the code has attempted to cast an object to a subclass of which it is not an instance". Cluster is not a subclass of JsonObject.